### DISCUSSION TITLE
Fantastic! Hope you can answer my questions

### DISCUSSION THREAD

**Discussion by dastrix80** [uhppoted/uhppoted#4] [Q&A] [answered]
Hi there,

Looking to build a RFID based door access for my home. 

Based off what I read I need a control board & reader

1 x https://de.aliexpress.com/item/4000781912427.html?spm=a2g0o.cart.0.0.f1823c00sODShF&mp=1

1 x https://www.ebay.com.au/itm/174766107007?chn=ps&norover=1&mkevt=1&mkrid=705-139619-5960-0&mkcid=2&itemid=174766107007&targetid=1280421173919&device=c&mktype=pla&googleloc=9071791&poi=&campaignid=12499395740&mkgroupid=124347745802&rlsatarget=pla-1280421173919&abcId=9300512&merchantid=116475353&gclid=Cj0KCQiAkNiMBhCxARIsAIDDKNUz2eNiE_Fp7FC0LVD6FBZj3gRs5re6CUyXdM8i9NnGhJtvH5kO9oQaAlrpEALw_wcB

I already have an existing Doorbird 125khz RFID tag which I think I can multi purpose for my Intercom and this door access solution.

Question re: software, the OEM software obviously has no REST API/HTTP integration. Am I right in saying the windows package of your software exposes the Control board via HTTP and that I could potentially send HTTP commands from a home automation package such as OpenHAB to control the electric strikes?

I would like to have the software programming done in Windows, but be able to send commands via HTTP or even MQTT to control the electric strikes

Thanks!

**Comment by uhppoted** [comment, marked as answer]
Sounds like a fun project! :-) 

Yup, you could use the REST or MQTT implementations - but TBH, I would really suggest starting with the [NodeRED](https://github.com/uhppoted/node-red-contrib-uhppoted) implementation:

- [NodeRED](https://nodered.org) is a really quick and easy and fun way to implement projects like this (especially when just trying to get up and running)
- The NodeRED standard library includes HTTP and MQTT implementations
- There are already a couple of NodeRED OpenHAB integrations e.g. [node-red-contrib-openhab](https://flows.nodered.org/node/node-red-contrib-openhab)

The _uhppoted-rest_ or _uhppoted-mqtt_ projects do expose the API via HTTP or MQTT respectively so you could of course hook your controller up to OpenHAB that way. But ..... they are probably overkill, more complicated than you probably need (at least initially) and would also require a lot more work on your side. They're really intended more for industrial'ish applications that live unloved in a dark cupboard somewhere.

Some other things you should be aware of:

- Based on a couple of other discussions, the AliExpress board is often an older version of the board (which may partially explain why it is a bit less expensive than e.g. the Amazon equivalent). At least one person I know of is using it successfully though - so it seems to be fine.
- Haven't ever used the HID 101E so can't really comment on that - it's apparently a 26-bit Wiegand interface so *should* work ok, but don't quote me on that. 
- There are multiple 125kHz card standards (which of course are not always compatible) so your Doorbird tag may not work with the HID. It's worth trying though - at worst you may have to carry two cards.

HTH (and please let me know how it goes!)

**Reply by dastrix80** [reply]
Thanks so much for the fast reply! I'll look into NodeRed, I only require basic functionality. Ive heard Nodered thrown around a bit but I've never looked into it - thanks for the tip! 

Will take your suggest and go for the Amazon board.

The Doorbird fob uses: ISO/IEC 18000-2:2009 Part 2, EM4100, EM4102

Shall do :)

**Comment by dastrix80** [comment]
Received my controller/RFID reader, I cant seem to locate a Windows Installer for this software? I read Windows was supported... Am I missing something? :(

**Comment by uhppoted** [comment]
The manufacturer software should have been supplied on a CD with the controller (the download site for it seems to have been taken offline or moved). If you didn't receive it you'll have to post a question on the Amazon page and ask their tech support for it.

The Windows binaries for this project can be found [here](https://github.com/uhppoted/uhppoted/releases/download/v0.7.1/uhppoted_v0.7.1-windows.zip) - it's just a zip file, no installer.

**Comment by dastrix80** [comment]
Yep I've got the CD, can use the controller just fine/on the network. Can use the CLI to open the door, that's all cool.

Intention , send MQTT command to the device to open the door relay.
